openedx/features/lti_course_tab/tab.py

```diff
@@ -0,0 +1,247 @@
+        return "{} - {}".format(
+            course.display_name_with_default,
+            course.display_org_with_default,
+        )
+
+    def _get_lti_config(self, course: CourseDescriptor) -> LtiConfiguration:
+        raise NotImplementedError
+
+    def _get_lti_embed_code(self, course: CourseDescriptor, request: HttpRequest) -> str:
+        """
+        Returns the LTI embed code for embedding in the current course context.
+        Args:
+            course (CourseDescriptor): CourseDescriptor object.
+            request (HttpRequest): Request object for view in which LTI will be embedded.
+        Returns:
+            HTML code to embed LTI in course page.
+        """
+        course_key = course.id
+        lti_config = self._get_lti_config(course)
+        lti_consumer = lti_config.get_lti_consumer()
+        user_id = quote(self._get_user_id(request.user, course_key))
+        context_id = quote(self._get_context_id(course_key))
+        resource_link_id = quote(self._get_resource_link_id(course_key, request))
+        roles = self._get_lti_roles(request.user, course_key)
+        context_title = self._get_context_title(course)
+        result_sourcedid = quote(self._get_result_sourcedid(context_id, resource_link_id, user_id))
+        additional_params = self._get_additional_lti_parameters(course, request)
+        locale = to_locale(get_language())
+
+        return lti_embed(
+            html_element_id='lti-tab-launcher',
+            lti_consumer=lti_consumer,
+            resource_link_id=resource_link_id,
+            user_id=user_id,
+            roles=roles,
+            context_id=context_id,
+            context_title=context_title,
+            context_label=context_id,
+            result_sourcedid=result_sourcedid,
+            launch_presentation_locale=locale,
+            **additional_params,
+        )
+
+    # pylint: disable=unused-argument
+    def render_to_fragment(self, request: HttpRequest, course: CourseDescriptor, **kwargs) -> Fragment:
+        """
+        Returns a fragment view for the LTI launch.
+        Args:
+            request (HttpRequest): request object
+            course (CourseDescriptor): A course object
+        Returns:
+            A Fragment that embeds LTI in a course page.
+        """
+        lti_embed_html = self._get_lti_embed_code(course, request)
+
+        fragment = Fragment(
+            HTML(
+                """
+                <iframe
+                    id='lti-tab-embed'
+                    srcdoc='{srcdoc}'
+                 >
+                </iframe>
+                """
+            ).format(
+                srcdoc=lti_embed_html
+            )
+        )
+        fragment.add_css(
+            """
+            #lti-tab-embed {
+                width: 100%;
+                min-height: 400px;
+                border: none;
+            }
+            """
+        )
+        return fragment
+
+
+class LtiCourseTab(LtiCourseLaunchMixin, EnrolledTab):
```

Comment:
There is still a lot of convenience we can add to this tab so it's usable for other tools, ideally a UI to add new tabs, but for now I think it serves the purpose.